conf/airframes/examples/ardrone2_gazebo_cyberzoo.xml

```diff
@@ -0,0 +1,213 @@
+<!DOCTYPE airframe SYSTEM "../airframe.dtd">
+
+<airframe name="ardrone2">
```

Comment:
Is the only difference here that gazebo loads the cyberzoo instead of the empty environment? Then I don't think you should add an extra example for this. One example specifically for gazebo is enough.

If you want to add an ardrone that can avoid orange poles that happens to also work in simulation then you can add this as ardrone2_pole_avoider or something like that. I still don't think that this should be in the examples but in the tudelft given that it is a bit specific to the cyberzoo.

Reply:
This will partially be solved with: https://github.com/paparazzi/paparazzi/pull/2172

But for now we still need a separate airframe file in order to have the orange_avoid module. One solution to this was: https://github.com/dewagter/paparazzi/tree/calibration_xml in which you could split airframe.xml into 2 parts: a common ardrone part and a few specific ones.

Reply:
I am not sure why you need a separate airframe. See https://github.com/paparazzi/paparazzi/blob/master/conf/airframes/tudelft/origami_lisamxs_wifi_indi_stereoboard.xml for an example of how to set up a single airframe for ap and nps.

https://github.com/paparazzi/paparazzi/blob/master/conf/airframes/KS/ks_bebop2_stereo.xml shows an example with a bebop.

I also don't see how #2172 solves your issue. That PR only makes it easier to set the default environment from the flightplan. Currently, this can already be done with the NPS_GAZEBO_WORLD define in the airframe file.

All you have to do is correctly define the nps target in https://github.com/paparazzi/paparazzi/blob/master/conf/airframes/tudelft/bebop_orangeavoid_course2017.xml then you can run it in simulation. 

As I mentioned before, the only possible issue is that the image might be a different size than on the real vehicle (the frame rate may also be different). The camera settings defined at the top of that airframe are not considered in nps yet, you will just get a fixed image with dimensions 320x240. This can be changed however, since #2158, sensors can now be added at runtime. So you could potentially make a bebop vehicle instead of the simple_x_quad which doesn't have cameras and add them at runtime with the correct settings. But none of this will be fixed by added a separate airframe.

Maybe if you describe why you need separate files in a bit more detail I can try to change nps to facilitate it.

Reply:
@kirkscheper the flying airframe is a bebop1, the simulated one is an ARDrone. Both have not only different board but different pretty much everything. Merging the control gains into 1 file is not working for me. I tried to have gazebo fly with bebop gains but that did not work either.